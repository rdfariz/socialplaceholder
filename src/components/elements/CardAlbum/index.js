import PropTypes from 'prop-types';
import { useHistory } from "react-router-dom";

function Component (props) {
  const { title, id, userId, className } = props;
  const history = useHistory();

  const handleClick = () => {
    history.push(`/albums/${id}`);
  };
  
  return (
    <div className={`${className} flex`}>
      <div className="w-full p-3 md:p-5 bg-white rounded-lg border cursor-pointer hover:shadow-md" onClick={handleClick}>
        <div className="text-sm font-medium">
          <p>Author ID: {userId}</p>
          <hr className="my-2" />
        </div>
        <h2 className="text-gray-800 text-lg font-semibold truncate ...">
          {title}
        </h2>
      </div>
    </div>
  )
}

Component.defaultProps = {
  title: '',
  id: '',
  userId: '',
  className: ''
};

Component.propTypes = {
  title: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  userId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string 
};

export default Component;
