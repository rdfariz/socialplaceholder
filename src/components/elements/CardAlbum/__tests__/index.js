import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CardAlbum from '../';
import { shallow } from 'enzyme';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('src/components/elements/CardAlbum', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <CardAlbum title="title" id="id" userId="10" />
    );
    expect(tree).toMatchSnapshot();
  });
  
  test('should return handleClick when card clicked', () => {
    const wrapper = shallow(<CardAlbum title="title" id="id" userId="10" />);
    const card = wrapper.find('.card_album');
    if (card.exists()) {
      card.first().simulate('click');
    }
  });
});
