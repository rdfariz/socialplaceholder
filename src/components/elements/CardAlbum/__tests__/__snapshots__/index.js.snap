// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardAlbum render 1`] = `
<div
  className=" flex"
>
  <div
    className="card_album w-full p-3 md:p-5 bg-white rounded-lg border cursor-pointer hover:shadow-md"
    onClick={[Function]}
  >
    <div
      className="text-sm font-medium"
    >
      <p>
        Author ID: 
        10
      </p>
      <hr
        className="my-2"
      />
    </div>
    <h2
      className="text-gray-800 text-lg font-semibold truncate ..."
    >
      title
    </h2>
  </div>
</div>
`;
